src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -171,5 +174,62 @@ public static SemanticMap GetSemanticMap(this SemanticModel semanticModel, Synta
 
             return new TokenSemanticInfo(declaredSymbol, aliasSymbol, allSymbols, type, convertedType, token.Span);
         }
+
+        public static string GenerateNameFromType(this SemanticModel semanticModel, ITypeSymbol type, ISyntaxFacts syntaxFacts, bool capitalize)
+        {
+            var pluralize = semanticModel.Pluralize(type);
```

Comment:
```suggestion
            var pluralize = semanticModel.ShouldPluralize(type);
```

Reply:
`Pluralize` makes it sound as if you're actually doing the work, not querying if it should be done.